setup.py

```diff
@@ -1,35 +1,62 @@
 #!/usr/bin/env python
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-# NOTE: The configuration for the package, including the name, version, and
-# other information are set in the setup.cfg file. Here we mainly set up
-# setup_requires and install_requires since these are determined
-# programmatically.
-
 import os
-import builtins
+import sys
+
+from setuptools import setup
+from extension_helpers import get_extensions
+
+# First provide helpful messages if contributors try and run legacy commands
+# for tests or docs.
+
+TEST_HELP = """
+Note: running tests is no longer done using 'python setup.py test'. Instead
+you will need to run:
+
+    tox -e test
+
+If you don't already have tox installed, you can install it with:
+
+    pip install tox
+
+If you only want to run part of the test suite, you can also use pytest
+directly with::
 
-import ah_bootstrap  # noqa
+    pip install -e .
+    pytest
 
-from astropy_helpers.distutils_helpers import is_distutils_display_option
-from astropy_helpers.setup_helpers import setup
+For more information, see:
 
-from setuptools.config import read_configuration
+  http://docs.astropy.org/en/latest/development/testguide.html#running-tests
+"""
 
-# We set up the following variable because we then use this in astropy/__init__.py
-# to make sure that we aren't importing astropy during the setup process (we used
-# to do this)
-builtins._ASTROPY_CORE_SETUP_ = True
+if 'test' in sys.argv:
+    print(TEST_HELP)
+    sys.exit(1)
 
-if is_distutils_display_option():
-    # Avoid installing setup_requires dependencies if the user just
-    # queries for information
-    setup_requires = []
-else:
-    setup_requires = read_configuration('setup.cfg')['options']['setup_requires']
-    # Make sure we have the packages needed for building astropy, but do not
-    # require them when installing from an sdist as the c files are included.
-    if not os.path.exists(os.path.join(os.path.dirname(__file__), 'PKG-INFO')):
-        setup_requires.extend(['cython>=0.29.13', 'jinja2>=2.7'])
+DOCS_HELP = """
+Note: building the documentation is no longer done using
+'python setup.py build_docs'. Instead you will need to run:
+
+    tox -e build_docs
+
+If you don't already have tox installed, you can install it with:
```

Comment:
Maybe give the alternative of
```
pip install -e .[docs]
cd docs
make html
```
here as well?

Reply:
Added